driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -172,6 +172,11 @@ public static ArrayExpression<StringExpression> ofStringArray(final String... ar
         return new MqlExpression<>((cr) -> new AstPlaceholder(new BsonArray(result)));
     }
 
+    public static <R extends DocumentExpression> R current() {
```

Comment:
There are times that users would need access to the current document as a `MapExpression`.  How would that be done?

(feel free to wait on this until after the map expression PR is merged)

Reply:
I think this is now possible via `current().isMapOr(ofMap(parse("{}")))`. This isn't great, but (while certainly possible) it seems very unusual for documents to be maps.

Reply:
It's not unusual for a document to be a `MapExpression<Expression>`. It always is.

Reply:
This is true in a certain sense, but I do not think that sense is relevant. For example, technically a Java object "is" (is representable as) a `Map<String, Object>`. But to use it as such, users must use the inconvenient reflection API.

A usage seems improbable if all of us have trouble imagining semi-realistic cases where users would have to store things in collections that need to be used as if they were maps. This would mean that they are storing things that have an actual identity (expressed in the `_id` field), and then some completely arbitrary fields (that is, the programmer cannot hard-code at least some of the other fields, and none of those fields were instead represented as a map on the thing). And this is happening often enough that the isMapOr conversion is burdensome.

Reply:
We discussed this and decided to go with inheritance (which would have been done in #1054), but then went back to non-inheritance due to an implementation problem that is ultimately caused by type erasure.

Added currentAsMap.